### PR TITLE
[ENH] include reviewers and committers in contributor list

### LINKED ISSUE #62: Credit multiple authors of PRs
### Description / Summary

Sometimes, PRs consist of commits from multiple authors, but github-activity only lists the user who opened the PR. It would be nice if the PR credit could include all commit authors (and possibly or optionally reviewers, as well).

### Value / benefit

This would improve credit for large PRs where multiple folks have contributed. Especially multi-stage PRs like the [jupyterhub RBAC one](https://github.com/jupyterhub/jupyterhub/pull/3438), or PRs where e.g. a maintainer re-issues a contributor's PR to finish it up.

### Implementation details

I think the API doesn't readily give access to all of a PR's authors, so you'll have to request all of the commits for the PR and iterate through them to retrieve the author and/or committer. A potential detail is that not all commits map to a GitHub user (e.g. due to a misconfigured git email address). I think those can be ignored, since they will ~always be the PR author or an auto-fix bot in my experience. But it will need to be handled to avoid an error, at least.

### Tasks to complete

- [ ] allow authors to be a list
- [ ] get authors from commits
- [ ] stretch: option to include reviewers, merged-by?

### LINKED ISSUE #67: Mention our definition of "contribution" in the output
### Context

We have a definition of "contributor" that includes discussion, opening ideas, providing reviews, etc. It is defined here:

https://github-activity.readthedocs.io/en/latest/#how-does-this-tool-define-contributions-in-the-reports

However, many people probably assume that "contributors" only means "code", and I think it is healthy for us to:

1. Be explicit about what we mean by contributing
2. Explicitly recognize non-code actions as "worthy" of contributor status

### Proposal

We should signal-boost our definition of "contributors" somehow. This might be as simple as adding an extra line to the `## Contributors to this release` output like:

> "The following people contributed discussions, new ideas, code and documentation contributions, and review. See [our definition of contributors](https://github-activity.readthedocs.io/en/latest/#how-does-this-tool-define-contributions-in-the-reports)."

Another way to resolve this would be to do:

- #68

### Tasks and updates

_No response_

### PR DESCRIPTION
**Pull request by minrk**
The most important part of this is done: getting the reviewer list and committer list from graphQL into the data.

- reviews, commits, and comments on merged PRs get counted in the 'all contributors' list
- Instead of only listing the PR author, each merged PR lists contributors to the PR in order:
  1. PR author
  2. all (unique) commit authors and committers
  3. who merged the PR
  4. reviewers
  5. commenters
  
  In practice, this means most PRs will have at least 2 names associated - the PR author and the merging maintainer.
- reviewers, committers, and commenters show up in the author list for each merged PR

new columns:

- reviews (raw graph)
- reviewers (unique list of reviewers)
- commits (raw graph)
- committers (ordered, non-unique list of committers)

I also moved creation of a couple derivative columns to the end of graphql from generate_activity_md, since that's where most similar code already is, and they were useful earlier in the code than they had been created.

Draft, because there are some open design questions, and I haven't touched any tests:

- [ ] do we separate 'author' (includes committers) from 'reviewers' (probably includes merged-by) in a visible way? If so, is it useful to distinguish merged-by from reviews?
- [ ] Should PR commenters be included; if so, should thresholds be applied, or is one comment sufficient?
- [ ] options - I think including multiple committers in the author list is strictly an improvement over what we have now with just the PR author. But should including reviewers/merger be behind an option, if so on/off by default, etc.? At what granularity (bool for each merger, reviewer, commenter)?

[sample output](https://gist.github.com/minrk/3c3f7a0dbf0c92b7c466544e50c32bf0) to compare with [jupyterhub 2.1.0](https://jupyterhub.readthedocs.io/en/2.1.0/changelog.html#id1).

closes #62
closes #67

### CODE REVIEW

**Comment by choldgraf** [comment]
I just want to quickly note that I think this would be a pretty cool and useful functionality to have :-)  (see #67)

**Comment by minrk** [comment]
I know it's been forever, but I've been using this to make the last several JupyterHub release notes, and I like it. I've updated the PR so it passes tests, at least.

I do think it warrants some options to expose to users to make different choices, and I don't know how folks would like to do that.

**Comment by choldgraf** [comment]
@minrk what do you think about just merging this in + cutting a release, and then iterating on configurability in future PRs? I feel like it is easier to add configurability rather than take it away.

**Comment by minrk** [comment]
@choldgraf it sounds fine to me, but of course I've entered everything _I_ want as the unconfigurable default, so I have no need of options :)

File: github_activity/github_activity.py [hunk 1]
```diff
@@ -60,6 +60,15 @@
     },
 }
 
+# exclude known bots from contributor lists
+# TODO: configurable? Everybody's got their own bots.
+BOT_USERS = {
+    "dependabot",
+    "meeseeksmachine",
+    "pre-commit-ci",
+    "welcome",
```
Comment (choldgraf) [inline github_activity/github_activity.py:69]:
```suggestion
    "names",
    "codecov",
    "codecov-io",
    "dependabot",
    "github-actions",
    "henchbot",
    "jupyterlab-dev-mode",
    "lgtm-com",
    "meeseeksmachine",
    "now",
    "pre-commit-ci",
    "renovate",
    "review-notebook-app",
    "support",
    "stale",
    "todo",
    "welcome",
```

Comment (choldgraf) [reply github_activity/github_activity.py:69]: a slightly bigger bot list name

**Review by choldgraf — APPROVED** [review]

**Comment by choldgraf** [comment]
Let's merge and iterate from there!

**Pull request merged**